### ISSUE TITLE
npm installation fails with `Error: No download link found`

### ISSUE THREAD

**Issue by NivedhaSenthil** [getgauge/gauge#1394]
### Expected behavior
npm install @getgauge/cli should work seamlessly

### Actual behavior
Fails sometime with 
```
Fetching download url for Gauge version 1.0.4
Error: No download link found for version 1.0.4, OS: darwin, Arch: x86_64
    at Request.request.get [as _callback] (/Users/vsts/agent/2.150.0/work/1/s/test/functional-tests/node_modules/@getgauge/cli/src/install.js:54:24)
    at Request.self.callback (/Users/vsts/agent/2.150.0/work/1/s/test/functional-tests/node_modules/request/request.js:185:22)
    at Request.emit (events.js:189:13)
    at Request.<anonymous> (/Users/vsts/agent/2.150.0/work/1/s/test/functional-tests/node_modules/request/request.js:1161:10)
    at Request.emit (events.js:189:13)
    at IncomingMessage.<anonymous> (/Users/vsts/agent/2.150.0/work/1/s/test/functional-tests/node_modules/request/request.js:1083:12)
    at Object.onceWrapper (events.js:277:13)
    at IncomingMessage.emit (events.js:194:15)
    at endReadableNT (_stream_readable.js:1125:12)
    at process._tickCallback (internal/process/next_tick.js:63:19)
```

### Steps to reproduce
npm install @getgauge/cli many times

seems to be happening because the url that is hit to get the download url has ratelimit https://api.github.com/repos/getgauge/gauge/releases/tags/v1.0.4

**Comment by negiDharmendra** [comment]
- First thing first. We also need to handle the error message and should not always assume that the download URL was not found for a given version.
https://github.com/getgauge/gauge/blob/577bef09d4edbe46c7bc79554039eb8a933ccdbb/build/npm/src/install.js#L44

- Second point. Github API has a [rate limit](https://developer.github.com/v3/#rate-limiting) of 60 request/hour for un-authenticated URLs.
There are a few options to overcome this issue.
    - Setup a proxy server to download the gauge binaries using [authenticated url](https://developer.github.com/v3/#increasing-the-unauthenticated-rate-limit-for-oauth-applications).
    - Or moving gauge binaries to a separate server.

- Third propagating proper message back to the user would be really helpful for debugging such issues.

**Comment by negiDharmendra** [comment]
For now, we would be just improving the error message.

**Comment by Debashis9012** [comment]
As per the above comment this issue has been verified and found fixed ```1.0.6.nightly-2019-05-17```
